CMakeLists.txt

```diff
@@ -95,7 +113,6 @@ set(UMD_HOME "${PROJECT_SOURCE_DIR}/tt_metal/third_party/umd")
 option(WITH_PYTHON_BINDINGS "Enables build of python bindings" ON)
 option(ENABLE_CODE_TIMERS "Enable code timers" OFF)
 option(ENABLE_TRACY "Enable Tracy Profiling" OFF)
-option(ENABLE_LIBCXX "Enable using libc++" ON)
```

Comment:
I checked locally.  The refinements during code review weren't enough to allow it to build without libc++.  So, yeah, that config was probably dead a while back.  Removing the option is more honest/transparent.